proposed/http-handlers/request-handlers-meta.md

```diff
@@ -363,6 +368,237 @@ executing that handler. If unable to route to a handler, it would instead
 execute the handler provided to the middleware. (This sort of mechanism can even
 be used in conjunction with middleware queues and stacks.)
 
+### 6.3 Example Interface Interactions
+
+The two interfaces, `RequestHandlerInterface` and `MiddlewareInterface`, were
+designed to work in conjunction with one another. While a
+`RequestHandlerInterface` can feasibly be used in isolation, it gains more power
+when used as a part of a middleware dispatch system. Middleware gains more
+flexibility by being de-coupled from any over-arching application layer, and
+instead only relying on the provided request handler to produce a response.
+
+Two approaches to middleware dispatch systems that the Working Group observed
+and/or implemented are demonstrated below. Additionally, examples of re-usable
+middleware are provided to demonstrate how to write middleware that is
+loosely-coupled.
+
+Please note that these are not suggested as definitive or exclusive approaches
+to defining middleware dispatch systems.
+
+#### Queue-based request handler
+
+In this approach, a request handler maintains a queue of middleware, and a
+fallback response to return if the queue is exhausted without returning a
+response. When executing the first middleware, the queue passes itself as a
+request handler to the middleware.
+
+```php
+class QueueRequestHandler implements RequestHandlerInterface
+{
+    private $middleware = [];
+    private $fallbackResponse;
+    
+    public function __construct(ResponseInterface $fallbackResponse)
+    {
+        $this->fallbackResponse = $fallbackResponse;
+    }
+    
+    public function add(MiddlewareInterface $middleware)
+    {
+        $this->middleware[] = $middleware;
+    }
+    
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Last middleware in the queue has called on the request handler.
+        if (0 === count($this->middleware)) {
+            return $this->fallbackResponse;
+        }
+        
+        $middleware = array_shift($this->middleware);
+        return $middleware->process($request, $this);
+    }
+}
+```
+
+An application bootstrap might then look like this:
+
+```php
+// Fallback response:
+$response = (new Response())->withStatus(500);
+
+// Create request handler instance:
+$app = new QueueResponseHandler($response);
+
+// Add one or more middleware:
+$app->add(new CorsMiddleware());
+$app->add(new RoutingMiddleware());
+
+// execute it:
+$response = $app->handle(ServerRequestFactory::fromGlobals());
+```
+
+This system has essentially two request handlers, one that will produce a
+response if no middleware generates one, and one for dispatching the middleware
+layers. (In this example, the `RoutingMiddleware` will likely execute composed
+handlers on a successful route match; see more on that below.)
```

Comment:
I don't really see 2 handlers here... I see one handler ($app) that will end up invoked 2-3 times.  Am I missing something or is this worded awkwardly?

Reply:
No, you're correct. The original example I created composed another request handler instead of a request prototype, and delegated to that handler if the last middleware in the queue calls on the handler instead of returning a response. I'm going to refactor the example to to do that, as it also then answers the "request prototype" question you have elsewhere.